microsoft/testsuites/performance/storageperf.py

```diff
@@ -187,6 +194,109 @@ def perf_storage_over_nfs_synthetic_tcp_4k(self, result: TestResult) -> None:
     def perf_storage_over_nfs_synthetic_udp_4k(self, result: TestResult) -> None:
         self._perf_nfs(result, protocol="udp")
 
+    @TestCaseMetadata(
+        description="""
+        This test case uses fio to test data disk performance with 4K block size.
+        """,
+        priority=3,
+        timeout=TIME_OUT,
+    )
+    def perf_storage_generic_fio_test(self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+        variables: Dict[str, Any],
+    ) -> None:
+        testcases = variables.get("fio_testcase_list", None)
+        if (not testcases):
+            testcases_file_path = os.path.join(
+                os.path.abspath(__file__).replace(os.path.basename(__file__), ""),
```

Comment:
`os.path.abspath(__file__).parent` is better. If it doesn't work try with pathlib.

Reply:
BTW, I feel it doesn't need a file. It can be put in code as dicts.

Reply:
We are using perf_disk function now with default parameters. This is resolved now.